pkg/nsx/cluster.go

```diff
@@ -124,6 +124,13 @@ func (cluster *Cluster) getThumbprint(addr string) string {
 func (cluster *Cluster) getCaFile(addr string) string {
 	host := addr[:strings.Index(addr, ":")]
 	var cafile string
+	// remove empty item from cluster.config.CAFile
```

Comment:
I remember @TaoZou1 resolved the issue here https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/config/config.go#L253
So could we leverage same way?

Reply:
But it seems it doesn't take effect, vpc_dev branch would report error

`2023-08-31 01:53:47.232	DEBUG	jwt/vcclient.go:212	creating vmomi client

2023-08-31 01:53:47.694	DEBUG	jwt/vcclient.go:274	generating certificate	{"user": "administrator@vsphere.local", "notBefore": "2023-08-31 01:47:47.694", "notAfter": "2023-08-31 02:53:47.694"}
2023-08-31 01:53:47.746	INFO	jwt/vcclient.go:85	creating new vapi session for vcClient
2023-08-31 01:53:47.909	DEBUG	jwt/tesclient.go:61	sending saml token to TES for JWT
2023-08-31 01:53:48.046	DEBUG	jwt/vcclient.go:236	HTTP request	{"request": "https://sc2-10-187-150-214.nimbus.eng.vmware.com:443/rest//vcenter/tokenservice/token-exchange", "response status": 200}
2023-08-31 01:53:48.047	DEBUG	jwt/tesclient.go:78	exchanged JWT
2023-08-31 01:53:48.055	LEVEL(-2)	util/utils.go:251	received HTTP response	{"response": "{\n  \"healthy\" : true,\n  \"components_health\" : \"MANAGER:UP, SEARCH:UP, UI:UP, NODE_MGMT:UP\"\n}"}
2023-08-31 01:53:48.055	INFO	nsx/endpoint.go:189	endpoint status is changing	{"endpoint": "10.187.157.154", "oldStatus": "DOWN", "newStatus": "UP"}
2023-08-31 01:53:48.055	INFO	nsx/endpoint.go:182	succeeded to setup endpoint
2023-08-31 01:53:48.062	LEVEL(-2)	util/utils.go:251	received HTTP response	{"response": "{\n  \"healthy\" : true,\n  \"components_health\" : \"MANAGER:UP, SEARCH:UP, UI:UP, NODE_MGMT:UP\"\n}"}
2023-08-31 01:53:48.444	LEVEL(-2)	util/utils.go:251	received HTTP response	{"response": "{\"node_version\": \"4.1.2.0.0.22234642\", \"product_version\": \"4.1.2.0.0.22234638\"}"}
2023-08-31 01:53:48.835	LEVEL(-2)	util/utils.go:251	received HTTP response	{"response": "{\"node_version\": \"4.1.2.0.0.22234642\", \"product_version\": \"4.1.2.0.0.22234638\"}"}
2023-08-31 01:53:48.836	INFO	nsx/cluster.go:133	CAFile	{"CAFile": [""]}
2023-08-31 01:53:48.836	ERROR	nsx/cluster.go:161	create transport	{"read ca file": "", "error": "open : no such file or directory"}
github.com/vmware-tanzu/nsx-operator/pkg/nsx.(*Cluster).createTransport.func1`

Reply:
Oh , I see, because the init is different.
config.NewNSXOperatorConfigFromFile vs config.NewNSXOpertorConfig()